WordPress/WordPressUITests/Screens/Editor/EditorScreen.swift

```diff
@@ -204,49 +213,61 @@ class EditorScreen: BaseScreen {
      */
     func addImageByOrder(id: Int) -> EditorScreen {
         tapToolbarButton(button: mediaButton)
-        let cameraRollButton = app.otherElements.cells["Camera Roll"]
-        waitFor(element: cameraRollButton, predicate: "isEnabled == true && isHittable == true")
-        cameraRollButton.tap()
 
-        // Wait for the Camera Roll Animation
-        let navigationBar = app.otherElements.navigationBars["Camera Roll"]
-        _ = navigationBar.waitForExistence(timeout: waitTimeout)
+        // Allow access to device media
+        app.tap() // trigger the media permissions alert handler
+
+        // Make sure media picker is open
+        if mediaButton.exists {
+            tapToolbarButton(button: mediaButton)
+        }
 
         // Inject the first picture
         app.cells.element(boundBy: 0).tap()
+        app.buttons["Insert 1"].tap()
 
-        // wait for upload simulation
-        sleep(6)
+        // Wait for upload to finish
+        _ = waitFor(element: uploadProgressBar, predicate: "exists == false", timeout: 10)
```

Comment:
One suggestion – you could use the `@discardableResult` annotation on `waitFor` to avoid needing `_ = ` for these sorts of things.

Reply:
Thanks, I wasn't aware of that!